### PR TITLE
Add Github CI

### PR DESCRIPTION
**Pull request by DLochmelis33**
The current CI fails on MacOS because of the new UPUB bug

### CODE REVIEW

File: .github/workflows/run-litmus.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        with:
+          distribution: oracle
+          java-version: 17
+      - run: chmod +x gradlew
+      - name: Assemble CLI binary
+        run: ./gradlew cli:linkReleaseExecutableLinuxX64
+      - name: Run litmus tests via CLI
+        run: ./cli/build/bin/linuxX64/releaseExecutable/cli.kexe -r pthread ".*"
+      - name: Run a single test with JCStress
+        run: ./gradlew :cli:jvmRun --args="-r jcstress -j '-m quick' StoreBuffering.Plain"
```
Comment (eupp) [inline .github/workflows/run-litmus.yml]: Why not run all jcstress tests?

Comment (eupp) [reply .github/workflows/run-litmus.yml]: @DLochmelis33 ping

Comment (eupp) [reply .github/workflows/run-litmus.yml]: Discussed this on a meeting. 

The problem is that apparently there is no option in jcstress to control the duration of testing, except the `-m` (mode) option. But, event with `-m quick` all the tests that we have now are run ~30mins, which is too long. 

The purpose of our local CI is to run "smoke-tests" of the `litmuskt` tool itself (i.e. to quickly check the basic functionality of the tool), and not to exhaustively test the Kotlin compiler.
Therefore, it was decided to run just a single jcstress test on CI, only to check that the jcstress integration works.

Comment (eupp) [reply .github/workflows/run-litmus.yml]: @DLochmelis33 could you please add the info from the comment above to the code. 
After this we can merge this PR.

**Comment by DLochmelis33** [comment]
Added the comment, decided to let JCStress run all tests, found a couple of bugs, fixed them

**Pull request merged**